wgsl/index.bs

```diff
@@ -3538,7 +4028,61 @@ compound_statement
   : BRACE_LEFT statements BRACE_RIGHT
 </pre>
 
-## Assignment TODO ## {#assignment}
+## Assignment ## {#assignment}
+
+An <dfn noexport>assignment statement</dfn> replaces the contents of a variable,
+or a portion of a variable, with a new value.
+
+The
+expression to the left of the equals token is the <dfn noexport>left-hand side</dfn>,
+and the
+expression to the right of the equals token is the <dfn noexport>right-hand side</dfn>.
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Statement<th>Description
+  </thead>
+  <tr algorithm="assignment">
+    <td>|r| : ref<|SC|,|T|>,<br>
+        |e| : |T|,<br>
+        |T| is [=storable=],<br>
+        |SC| is a writable [=storage class=]
+    <td class="nowrap">|r| = |e|;
+    <td>Evaluates |e|, evaluates |r|, then replaces the contents of memory referenced by |r| with the value computed for |e|.<br>
+        (OpStore)
+</table>
+
+The [=originating variable=] of the left-hand side must not have an `access(read)` access attribute.
+
+In the simplest case, the left hand side of the assignment statement is the
+name of a variable.  See [[#forming-a-reference-value]] for other cases.
+
+    <div class='example wgsl' heading='Assignments'>
+      <xmp highlight='rust'>
+        struct S {
+            age: i32;
+            weight: f32;
+        };
+        var<private> person : S;
+
+        fn f() {
+            var a: i32 = 20;
+            a = 30;           // Replace the contents of 'a' with 30.
+
+            person.age = 31;  // Write 31 into the age field of the person variable.
+
+            var uv: vec2<f32>;
+            uv.y = 1.25;      // Place 1.25 into the second component of uv.
+
+            const uv_x_reference: ref<function,f32> = uv.x;
```

Comment:
This should be removed when you revisit.

Reply:
I changed it to use pointers.